src/test/java/bdd/steps/NameMatchingSteps.java

```diff
@@ -231,79 +232,148 @@ public void theFootprintWillTryTheFollowingCombinationOfNamesInOrder(DataTable d
         return aResponse()
-                .withStatus(HttpStatus.FORBIDDEN.value())
+                .withStatus(SC_FORBIDDEN)
                 .withBody("{ \"code\":\"MATCHING_FAILED\",\"message\":\"There is no match for the information provided\"}")
                 .withHeader("Content-Type", MediaType.APPLICATION_JSON_VALUE);
     }
 
     @And("^HMRC was called (\\d+) times$")
     public void hmrcWasCalledTimes(int numberOfCalls) {
         List<LoggedRequest> matchingRequests = getIndividualMatchingRequestsInOrder();
-        assertEquals(numberOfCalls, matchingRequests.size());
+        assertThat(matchingRequests.size()).isEqualTo(numberOfCalls);
     }
 
     @Then("^a not matched response is returned$")
     public void aNotMatchedResponseIsReturned() {
         Response response = getIndividualMatchingResponse();
 
-        assertNotNull(response);
-        assertEquals(404, response.getStatusCode());
+        assertThat(response).isNotNull();
+        assertThat(response.getStatusCode()).isEqualTo(SC_NOT_FOUND);
+    }
+
+
+    @Then("^the following words will be used for the first part of the last name$")
+    public void theFollowingWordsWillBeUsedForTheFirstPartOfTheLastName(DataTable dataTable) {
+
+        List<String> lastNames = dataTable.asList(String.class);
+        long matches = numberOfMatchingRequests(lastNames, this::jsonContainingLastNameOf);
+
+        assertThat(matches).isEqualTo(lastNames.size());
+    }
+
+    @Then("^the following names will be tried$")
+    public void theFollowingNamesWillBeTried(DataTable dataTable) {
+
+        List<IndividualRow> names = dataTable.asList(IndividualRow.class);
+        long matches = numberOfMatchingRequests(names, this::jsonContainingAllNamesOf);
+
+        assertThat(matches).isEqualTo(names.size());
+    }
+
+    @Then("^the following names will not be tried$")
+    public void theFollowingNamesWillNotBeTried(DataTable dataTable) {
+
+        List<IndividualRow> names = dataTable.asList(IndividualRow.class);
+        long matches = numberOfMatchingRequests(names, this::jsonContainingAllNamesOf);
+
+        assertThat(matches).isEqualTo(0);
+    }
+
+    private <T> long numberOfMatchingRequests(List<T> names, Function<T, String> produceJsonForName) {
+
+        List<LoggedRequest> loggedRequests = getIndividualMatchingRequestsInOrder();
+
+        return names
+                   .stream()
+                   .map(produceJsonForName)
+                   .map(this::produceExpectedNameMatcher)
+                   .map(expectedNameMatcher -> loggedRequests
+                                                         .stream()
+                                                         .map(LoggedRequest::getBodyAsString)
+                                                         .anyMatch(requestBody -> expectedNameMatcher.match(requestBody).isExactMatch()))
+                   .filter(this::aRequestWasMadeForThisName)
```

Comment:
As we have discussed `.filter(m -> m == true)` is probably the clearest way to express this.